src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        result.push({ ...stateCopy });
+
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        result.push({ ...stateCopy });
+
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        result.push({ ...stateCopy });
```

Comment:
since you've written a new object literal no need to copy it via the spread operator - it is already an entirely new object, so you can do it in this way: `push(stateCopy);`

Reply:
If I don't use the spread operator in this case the last test will fail. Because in case 'addProperties ' after Object.assign the last object, which was pushed after case 'clear' will be changed too.
![image](https://user-images.githubusercontent.com/98056446/209545514-572aade6-ada9-4def-8379-e5906eb01693.png)
![image](https://user-images.githubusercontent.com/98056446/209545547-e186bacb-42ae-41e9-8328-72263db2b35b.png)
![image](https://user-images.githubusercontent.com/98056446/209545568-25bcd728-5d7d-4002-a49d-8ff9b2321128.png)